auth_service/auth_adapter/api/basic.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2021 - 2022 Universität Tübingen, DKFZ and EMBL
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Basic HTTP authentication
+"""
+
+import secrets
+
+from fastapi import Depends, FastAPI, HTTPException, status
+from fastapi.responses import JSONResponse, PlainTextResponse
+from fastapi.security import HTTPBasic, HTTPBasicCredentials
+
+from .deps import get_config
+
+__all__ = ["basic_auth_injector"]
+
+
+def basic_auth_injector(app: FastAPI):
+    """Inject Basic authentication if user and password are set."""
+    config = get_config()
+    user, pwd = config.basic_auth_user, config.basic_auth_pwd
+    if not (user and pwd):
+        return None
+
+    security = HTTPBasic(realm="GHGA Data Portal")
```

Comment:
Would it make sense to set the realm in the config class?

Reply:
At least it cannot harm, did so.